### ISSUE TITLE
T-net code problem

### ISSUE THREAD

**Issue by public1024** [charlesq34/pointnet#109]
Hi, Im quite new to tensorflow. I read your code and notice here:
`    with tf.variable_scope('transform_feat') as sc: 
        weights = tf.get_variable('weights', [256, K*K],
                                  initializer=tf.constant_initializer(0.0),
                                  dtype=tf.float32)
        biases = tf.get_variable('biases', [K*K],
                                 initializer=tf.constant_initializer(0.0),
                                 dtype=tf.float32)
        biases += tf.constant(np.eye(K).flatten(), dtype=tf.float32)
        transform = tf.matmul(net, weights)
        transform = tf.nn.bias_add(transform, biases)
`
if u initialize "weights" with 0, should not it stay 0 for ever?

**Comment by LoFaiTh** [comment]
I have the same question. Have any one figure this out? Can you explain this to us? @charlesq34

**Comment by ameliajimenez** [comment]
Hi, no, it won't. 

That's just the initialization of the weights of that layer. They will be updated after computing the gradients of each batch, like any other weights from convolutional layers.
